tests/testthat/test-get_source_expressions.R

```diff
@@ -192,3 +192,31 @@ test_that("1- or 2-width octal expressions give the right STR_CONST values", {
     expect_identical(pc[[1L]][8L, "text"], "'\n\\2\n'")
   })
 })
+
+test_that("linters pass with xml_missing() content", {
+  # NB: this is just a cursory test for linters not to
+  #   fail on files where the XML content is xml_missing;
+  #   the main linter test files provide more thorough
+  #   evidence that things are working as intended.
+  bad_source <- withr::local_tempfile()
+  writeLines("a = 1\nb = 2", bad_source)
+  expressions <- get_source_expressions(bad_source)$expressions
+
+  # "zap" the xml_parsed_content to be xml_missing -- this gets
+  #   around the issue of creating a file that fails to parse now,
+  #   but later fails in a different way -> xml not missing.
+  for (ii in seq_along(expressions)) {
+    if ("xml_parsed_content" %in% names(expressions[[ii]])) {
+      expressions[[ii]]$xml_parsed_content <- xml2::xml_missing()
+    } else {
+      expressions[[ii]]$full_xml_parsed_content <- xml2::xml_missing()
+    }
+  }
+
+  # test all linters at least pass (no error) when finding xml_missing
+  for (linter in linters_with_tags(tag = NULL)) {
+    for (expr in expressions) {
+      expect_error(linter(expr), NA)
```

Comment:
Wouldn't we want to tighten this to `expect_silent()` at least for our own linters?
Also, I can imagine the failure message would be awful to debug. Is this a usecase for {patrick}?

Reply:
`expect_silent()` doesn't quite capture it (since the `a = 1` didn't throw a failure for `assignment_linter()`).

`expect_length(., 0L)` fits the bill better, but because we're using a bit of an awkward interface to lint, we have to use the workaround to #1160 -- `nrow(as.data.frame())`.

Reply:
Oh I think we need to check both, i.e. no output and no lint, then? I'd say we should make our linters consistently return `list()` and the use the simpler test.

Reply:
from typical user perspective it's the same because `flatten_lints()` gives the same output... so i think we're better off leaving that part as a separate fix in #1160.

Reply:
I think we want `lint()` to be silent, i don't think that works with the current test setup.

Reply:
> from typical user perspective it's the same because `flatten_lints()` gives the same output... so i think we're better off leaving that part as a separate fix in #1160.

I agree it should be a separate PR, just think we should fix it before merging this one so we get a nicer test from the start.

Reply:
> I think we want `lint()` to be silent, i don't think that works with the current test setup.

I'm just hoping to catch any stray warnings that may or may not be produced by linters when they encounter `xml_missing()` for now.

Reply:
I think we want expect_warning(., NA) then

Reply:
> just think we should fix it before merging this one so we get a nicer test from the start.

I think it's fine, the test is only a little big uglier for now. would agree if it was on the R side.

the main thing is I think the fix will take some effort. don't see a need to hold up this PR (the last code change of the release for now) while we fix internals

Reply:
Ah thanks for pointing this out. I didn't realize it was that much work.

Is `expect_length(flatten_lints(...), 0L)` an option?

Reply:
flatten_lints not exported right? it'll work but I'm usually hesitant to use unexported stuff. here's probably an ok case as an expedient... is it that much better than nrow(as.df())?

Reply:
> I didn't realize it was that much work.

yea i think it needs a refactor.

current structure is `lapply(foo, function(x) if (bar) then_lint)`. we have to break the inner `if` outside the `lapply` IINM.

Reply:
It does seem more concise to me, and even more so with a wrapping `expect_warning(..., NA)`...

Fine with either approach as long as we test for warning-free and lint-free results though.

Reply:
OK, I think it should be ready to go